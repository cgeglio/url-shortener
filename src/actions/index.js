export const setUrls = (urls) => ({
  type: 'SET_URLS',
  urls
});

export const addUrl = (url) => ({
  type: 'ADD_URL',
  url
});

export const deleteUrl = (url) => ({
  type: 'DELETE_URL',
  url
});
