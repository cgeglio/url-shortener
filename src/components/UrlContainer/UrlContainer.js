import React, { Component } from 'react';
import './UrlContainer.css';
import { connect } from 'react-redux';
import { setUrls } from '../../actions';
import { getUrls } from '../../apiCalls';

export class UrlContainer extends Component {

  componentDidMount() {
    getUrls()
      .then(data => this.props.setUrls(data.urls))
      .catch(err => console.error('Error fetching:', err));
  }

  render() {
    return (
      !this.props.urls.length ? <p>No urls yet! Find some to shorten!</p> :
      <section>
        {this.props.urls.map(url =>
          <div className="url">
            <h3>{url.title}</h3>
            <a href={url.short_url} target="blank">{url.short_url}</a>
            <p>{url.long_url}</p>
          </div>
        )}
      </section>
    )
  }
}

export const mapStateToProps = state => ({
  urls: state.urls
});

export const mapDispatchToProps = dispatch => ({
  setUrls: urls => dispatch(setUrls(urls))
});

export default connect(mapStateToProps, mapDispatchToProps)(UrlContainer);
