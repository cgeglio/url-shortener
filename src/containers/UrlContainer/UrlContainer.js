import React, { Component } from 'react';
import './UrlContainer.css';
import { connect } from 'react-redux';
import { setUrls, deleteUrl } from '../../actions';
import { getUrls, removeUrl } from '../../apiCalls';

export class UrlContainer extends Component {

  componentDidMount() {
    getUrls()
      .then(data => this.props.setUrls(data.urls))
      .catch(err => console.error('Error fetching:', err));
  }

  removeCurrentUrl = (url) => {
    removeUrl(url.id)
      .then(() => this.props.deleteUrl(url))
      .catch(error => console.log('Error deleting url'))
  }

  render() {
    return (
      !this.props.urls.length ? <p>No urls yet! Find some to shorten!</p> :
      <section>
        {this.props.urls.map(url =>
          <div className="url">
            <h3>{url.title}</h3>
            <a href={url.short_url} target="blank">{url.short_url}</a>
            <p>{url.long_url}</p>
            <button onClick={() => this.removeCurrentUrl(url)}>Delete Me!</button>
          </div>
        )}
      </section>
    )
  }
}

export const mapStateToProps = state => ({
  urls: state.urls
});

export const mapDispatchToProps = dispatch => ({
  setUrls: urls => dispatch(setUrls(urls)),
  deleteUrl: url => dispatch(deleteUrl(url))
});

export default connect(mapStateToProps, mapDispatchToProps)(UrlContainer);
