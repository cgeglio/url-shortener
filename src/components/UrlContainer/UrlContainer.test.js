import { UrlContainer, mapStateToProps, mapDispatchToProps } from './UrlContainer';
import React from 'react';
import { setUrls } from '../../actions';
import { shallow } from 'enzyme';

describe('UrlContainer', () => {
  it('should match the snapshot', () => {
    let wrapper = shallow(<UrlContainer urls={[]}/>)
    expect(wrapper).toMatchSnapshot();
  })
})

describe('mapStateToProps', () => {
  it('should return an array of url objects', () => {
    const mockState = {
      urls: [{
        id: 1,
        long_url: "https://images.unsplash.com/photo...",
        short_url: "http://localhost:3001/useshorturl/2",
        title: 'Awesome photo'
      }],
      cats: [{name: 'Bithcuits'}, {name: 'Peanut'}]};

    const expected = {
      urls: [{
        id: 1,
        long_url: "https://images.unsplash.com/photo...",
        short_url: "http://localhost:3001/useshorturl/2",
        title: 'Awesome photo'
      }]
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  })
});

describe('mapDispatchToProps', () => {
  it('should call dispatch with the setUrls action when setUrls is called', () => {
    const mockUrls = [{
      id: 1,
      long_url: "https://images.unsplash.com/photo...",
      short_url: "http://localhost:3001/useshorturl/2",
      title: 'Awesome photo'
    }];
    const mockDispatch = jest.fn();
    const actionToDispatch = setUrls(mockUrls);
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.setUrls(mockUrls);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  })
});
