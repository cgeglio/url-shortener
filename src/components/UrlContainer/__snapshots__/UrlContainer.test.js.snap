// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlContainer should match the snapshot 1`] = `
<p>
  No urls yet! Find some to shorten!
</p>
`;
